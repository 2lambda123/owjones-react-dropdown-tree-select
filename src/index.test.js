import test from 'ava'
import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import DropdownTreeSelect from './index'

let tree

test.beforeEach(() => {
  tree = [
    {
      label: 'item1',
      value: 'value1',
      children: [
        {
          label: 'item1-1',
          value: 'value1-1',
          children: [{ label: 'item1-1-1', value: 'value1-1-1' }, { label: 'item1-1-2', value: 'value1-1-2' }]
        },
        { label: 'item1-2', value: 'value1-2' }
      ]
    },
    {
      label: 'item2',
      value: 'value2',
      children: [
        {
          label: 'item2-1',
          value: 'value2-1',
          children: [
            { label: 'item2-1-1', value: 'value2-1-1' },
            { label: 'item2-1-2', value: 'value2-1-2' },
            {
              label: 'item2-1-3',
              value: 'value2-1-3',
              children: [{ label: 'item2-1-3-1', value: 'value2-1-3-1' }]
            }
          ]
        },
        { label: 'item2-2', value: 'value2-2' }
      ]
    }
  ]
})

test('renders default state', t => {
  const wrapper = shallow(<DropdownTreeSelect data={tree} />)
  t.snapshot(toJson(wrapper))
})

test('shows dropdown', t => {
  const wrapper = shallow(<DropdownTreeSelect data={tree} showDropdown />)
  t.snapshot(toJson(wrapper))
})
