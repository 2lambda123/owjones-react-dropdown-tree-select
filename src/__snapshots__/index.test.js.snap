// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownTreeSelect/> renders correctly 1`] = `
<div
  className="react-dropdown-tree-select"
>
  <div
    className="dropdown"
  >
    <a
      className="dropdown-trigger arrow bottom"
      onClick={[Function]}
    >
      <ul
        className="tag-list"
      >
        <li
          className="tag-item"
        >
          <input
            className="search"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Choose..."
            type="text"
          />
        </li>
      </ul>
    </a>
  </div>
</div>
`;
